es-app/src/SystemData.cpp

```diff
@@ -365,7 +366,7 @@ SystemData* SystemData::getNext() const
 		it++;
 		if (it == sSystemVector.cend())
 			it = sSystemVector.cbegin();
-	} while ((*it)->getDisplayedGameCount() == 0); 
+	} while ((*it)->getDisplayedGameCount() == 0 && !(UIModeController::getInstance()->isUIModeFull() && (*it)->isCollection()));
```

Comment:
So, the plain-language version of this code is: while the system aught to be hidden, go to the next / previous system.
In the past, this was a simple criterion: if there is anything to show, it should not be hidden, Now however, its a bit more complicated, (and might even be more complicated in the future).
In any case, I think that this method should have less knowledge on these details, but rather should be able to ask: "should this system be visible now?", So I would propose to create a new method: 
```
bool isVisible()
{
   return (getDisplayedGameCount() > 0 || 
           !(UIModeController::getInstance()->isUIModeFull() && (*it)->isCollection()));
}
```
That same method can then be called by getPrevious(), and during population (SystemView line 40).